core/charm/repository/charmhub_test.go

```diff
@@ -245,11 +245,15 @@ func (s *charmHubRepositorySuite) TestResolveRevisionNotFoundErrorWithNoSeries(c
 		Platform: corecharm.Platform{
 			Architecture: arch.DefaultArchitecture,
 		},
+		Channel: &charm.Channel{},
 	}
 
 	repo := NewCharmHubRepository(s.logger, s.client)
 	_, _, _, err := repo.ResolveWithPreferredChannel(curl, origin, nil)
-	c.Assert(err, gc.ErrorMatches, `selecting releases: no charm or bundle matching channel or platform; suggestions: stable with focal`)
+	c.Assert(err, gc.ErrorMatches,
+		`(?m)selecting releases: no charm or bundle matching channel "" or platform "amd64"
+suggestions:
```

Comment:
I'd suggest (no pun intended) to leave out the "suggestions:" line altogether. I think it's pretty clear that the subsequent lines are suggestions. Or failing that, move "; suggestions:" to the end of the previous line and only have the suggestions on new lines. But get someone else's take too.